packages/chatbot-server-mongodb-public/src/eval/bin/generateEvalCasesYamlFromCSV.ts

```diff
@@ -89,13 +124,30 @@ async function main({
     csvFilePath,
     transformationType ? transformationMap[transformationType] : undefined
   );
+  const expectedSources = new Set<string>();
+  evalCases.forEach((caseItem) => {
+    caseItem.expectedLinks?.forEach((url) => {
+      expectedSources.add(url);
+    });
+  });
+  const urlList = Array.from(expectedSources);
```

Comment:
```suggestion
  const urlList = Array.from(
    new Set(evalCases.flatMap((caseItem) => caseItem.expectedLinks ?? [])
  );
```

Potential here for `Array.prototype.flatMap()` instead of nested `Array.prototype.forEach()`

Reply:
Thanks for the suggestion, much cleaner solution